pubtools/exodus/_tasks/push.py

```diff
@@ -83,7 +83,10 @@ def run(self):
                 stderr=subprocess.STDOUT,
                 universal_newlines=True,
             )
-            for line in proc.stdout:
+
+            for line in iter(proc.stdout.readline, b""):
+                if line == "":
+                    break
```

Comment:
Seems like this code will get the right result, but I think it's slightly overcomplicated.

That `iter(proc.stdout.readline, b"")` is already supposed to make it stop when it reaches the end, right?  So why do we also need to explicitly check for an empty value inside of the loop?  We shouldn't need two separate checks to see if we've hit the end of the stream.

I think the reason is that you used `b""` (empty byte string) instead of `""` there.  So you're asking iter to stop if it finds an empty *byte* string and then you're explicitly checking for an empty *text* string.

This shouldn't be necessary.  The Popen was created with universal_newlines=True which means it'll always give you text, not bytes, on both py2 and py3.  I think the situation then is:

- on python3, the iter with `b""` never stops because readline at end returns `""`, and `b"" == ""` is false on python3.  So it needs to use the `if line == ""` check within the loop.
- on python2, the iter with `b""` stops because readline at end returns `""`, and `b"" == ""` is true on python2.  So the `if line == ""` check within the loop is never reached.

It seems unnecessarily complicated/sloppy and it ought to instead be possible to just write this:

```suggestion
            for line in iter(proc.stdout.readline, ""):
```

Or if that doesn't work, I'd be interested in learning why.